scene/2d/tile_map.cpp

```diff
@@ -1782,6 +1785,16 @@ bool TileMap::get_clip_uv() const {
 	return clip_uv;
 }
 
+void TileMap::set_texture_filter(TextureFilter p_texture_filter) {
+	CanvasItem::set_texture_filter(p_texture_filter);
+	_recreate_quadrants();
```

Comment:
Why are you recreating quadrants here? You can probably just updated the texture filter directly in the server. Same goes for below.

Reply:
Can you give me an advice of how to do that?

Reply:
Well, I'm just guessing, but it seems odd to have to re create the quadrants just to change a setting. 

Are you able to retrieve the RIDs for the quadrants? If so, you could just loop through them and set the filter mode with the same line you included above 
``vs->canvas_item_set_default_texture_filter(canvas_item, VS::CanvasItemTextureFilter(CanvasItem::get_texture_filter()))``

Reply:
I tried with this, but is not working, I don't know exactly what i'm missing.

```
void TileMap::_update_textures() {
	for (Map<PosKey, Quadrant>::Element *F = quadrant_map.front(); F; F = F->next()) {
		Quadrant &q = F->get();
		for (List<RID>::Element *E = q.canvas_items.front(); E; E = E->next()) {
			RenderingServer::get_singleton()->canvas_item_set_default_texture_filter(E->get(), RS::CanvasItemTextureFilter(CanvasItem::get_texture_filter()));
			RenderingServer::get_singleton()->canvas_item_set_default_texture_repeat(E->get(), RS::CanvasItemTextureRepeat(CanvasItem::get_texture_repeat()));
		}
	}
}
```

Reply:
Where are you calling that?

Reply:
I resolve merge conflicts and force push with the `_recreate_quadrants();` if someone want to test it, I will continue searching tomorrow.

I don't know how to update this properly and I'm not having such time, if someone can help with this will be nice. I think you're correct clayjohn but I don't know exactly how to do it.

Reply:
> Where are you calling that?

In set_texture_filter and set_texture_repeat

```void TileMap::set_texture_filter(TextureFilter p_texture_filter) {
	CanvasItem::set_texture_filter(p_texture_filter);
	_update_textures();
}```

Reply:
Here is a slightly improved approach. This sets all quadrants as dirty and refreshes them instead of completely clearing all quadrants and rebuilding them from scratch. Although, this still seems like we are doing way too much work just to update the texture filter. 

```
void TileMap::set_texture_filter(TextureFilter p_texture_filter) {
	CanvasItem::set_texture_filter(p_texture_filter);
	for (Map<PosKey, Quadrant>::Element *F = quadrant_map.front(); F; F = F->next()) {
		Quadrant &q = F->get();
		for (List<RID>::Element *E = q.canvas_items.front(); E; E = E->next()) {
			RenderingServer::get_singleton()->canvas_item_set_default_texture_filter(E->get(), RS::CanvasItemTextureFilter(p_texture_filter));
			_make_quadrant_dirty(F);
		}
	}
}

void TileMap::set_texture_repeat(CanvasItem::TextureRepeat p_texture_repeat) {
	CanvasItem::set_texture_repeat(p_texture_repeat);
	for (Map<PosKey, Quadrant>::Element *F = quadrant_map.front(); F; F = F->next()) {
		Quadrant &q = F->get();
		for (List<RID>::Element *E = q.canvas_items.front(); E; E = E->next()) {
			RenderingServer::get_singleton()->canvas_item_set_default_texture_repeat(E->get(), RS::CanvasItemTextureRepeat(p_texture_repeat));
			_make_quadrant_dirty(F);
		}
	}
}
```

Reply:
That's working, maybe there is another way to update the texture filter, but I tried to figure out a way to do better without success.
I force-push with your code anyway.

Reply:
I asked reduz and he said is ok